setup.cfg

```diff
@@ -3,3 +3,11 @@ test=pytest
 
 [tool:pytest]
 addopts = --ignore=scripts
+
+[flake8]
+ignore = E501, W503
+max-line-length = 79
+# it's recommended to have max-complexity ~ 18
+max-complexity = 30
+# should probably also have "F" here -- bad imports and variable definitions
+select = B,C,E,W,T4,B9
```

Comment:
what is complexity a measure of? What do these things in select mean? Should F be added?

Reply:
explicitly, this: https://en.wikipedia.org/wiki/Cyclomatic_complexity

Basically, you want to keep complexity down to make the code have less places to go wrong. I think the default is about 18. `flake8` will error if the complexity is higher than `max-complexity`. The highest in `pyuvsim` is 30 (so I set it to that to get it to pass). I figure reducing it should be another PR. 

The "Select" identifiers give various kinds of formatting errors/bad practices that should be caught. Error codes are mostly from `pycodestyle`: https://pycodestyle.readthedocs.io/en/latest/intro.html#error-codes, but "F" codes are new in `flake8`: http://flake8.pycqa.org/en/latest/user/error-codes.html, and the "C" code is for complexity. 

In particular, "F" catches a lot of stuff like "module imported but not used". Lots of things fail if we also select for "F", which I think should be fixed, but probably in a separate PR.

Reply:
That's very interesting, and looks to be a useful metric. Do you think the current complexity may be coming from the number of setup options we have? Once the simulation starts, there's pretty much just one path it follows.

Reply:
Ok, sounds reasonable. Reading through the wikipedia page suggests complexity can be measured at different levels. Is 30 a measure for the entire package or for individual files or something else?

Reply:
that 30 was for some individual _functions_ . Setting `max-complexity=18`, I get the following output:
```
./pyuvsim/simsetup.py:315:1: C901 'create_mock_catalog' is too complex (21)
./pyuvsim/simsetup.py:557:1: C901 'parse_telescope_params' is too complex (30)
./pyuvsim/simsetup.py:709:1: C901 'parse_frequency_params' is too complex (20)
./pyuvsim/simsetup.py:961:1: C901 'initialize_uvdata_from_params' is too complex (21)
./pyuvsim/simsetup.py:1093:1: C901 'initialize_uvdata_from_keywords' is too complex (19)
./pyuvsim/utils.py:182:1: C901 'write_uvdata' is too complex (19)
```

Reply:
I propose we create an issue for this and move past it in this PR.

Reply:
Yes, I agree. I'm happier that it's a measure of functions. I agree those should be more simple, I was worried if it was for too large a unit of code (which might need to be complex to work properly).

Reply:
It makes sense that those functions are complex. They handle several many different setup options. I'm a little surprised that `parse_frequency_params` is too complex but `parse_time_params` isn't.